corehq/apps/userreports/templates/userreports/edit_data_source.html

```diff
@@ -2,100 +2,157 @@
 {% load i18n %}
-
 {% js_entry_b3 'userreports/js/widgets' %}
 
 {% block page_content %}
   {% if data_source.meta.build.is_rebuild_in_progress %}
-        <div id="built-warning" class="alert alert-warning">
-          <h4><i class="fa-solid fa-triangle-exclamation"></i>
-          {% if data_source.rebuild_failed %}
-            {% trans "Populating your report did not complete successfully." %}
-          {% elif data_source.meta.build.is_rebuilding %}
-            {% trans "Please note that this datasource is being rebuilt." %}
-          {% elif data_source.meta.build.is_rebuilding_in_place %}
-            {% trans "Please note that this datasource is being rebuilt in place." %}
-          {% endif %}
-          </h4>
-          {% trans "If rebuilt again next rebuild will only start when the previous rebuild(s) finishes." %}
-        </div>
+    <div id="built-warning" class="alert alert-warning">
+      <h4>
+        <i class="fa-solid fa-triangle-exclamation"></i>
+        {% if data_source.rebuild_failed %}
+          {% trans "Populating your report did not complete successfully." %}
+        {% elif data_source.meta.build.is_rebuilding %}
+          {% trans "Please note that this datasource is being rebuilt." %}
+        {% elif data_source.meta.build.is_rebuilding_in_place %}
+          {% trans "Please note that this datasource is being rebuilt in place." %}
+        {% endif %}
+      </h4>
+      {% trans "If rebuilt again next rebuild will only start when the previous rebuild(s) finishes." %}
+    </div>
   {% endif %}
   {% if data_source.get_id %}
     <div class="btn-toolbar pull-right">
+      {% if use_updated_ucr_naming %}
         <div class="btn-group">
-          {% if use_updated_ucr_naming %}
-            <a href="{% url 'preview_configurable_data_source' domain data_source.get_id %}" class="btn btn-default">{% trans 'Preview Custom Web Report Data' %}</a>
-            <a href="{% url 'summary_configurable_data_source' domain data_source.get_id %}" class="btn btn-default">{% trans 'Custom Web Report Source Summary' %}</a>
-          {% else %}
-            <a href="{% url 'preview_configurable_data_source' domain data_source.get_id %}" class="btn btn-default">{% trans 'Preview Data' %}</a>
-            <a href="{% url 'summary_configurable_data_source' domain data_source.get_id %}" class="btn btn-default">{% trans 'Data Source Summary' %}</a>
+          <a
+            href="{% url 'preview_configurable_data_source' domain data_source.get_id %}"
+            class="btn btn-default {% if not data_source_table_exists %}disabled{% endif %}"
+            >{% trans 'Preview Custom Web Report Data' %}</a
+          >
+          {% if not data_source_table_exists %}
+            <span
+              class="hq-help-template"
+              data-title="{% trans_html_attr 'Preview Unavailable' %}"
+              data-content="{% trans_html_attr 'Preview unavailable till the data source table has been created.' %}"
```

Comment:
This was a fun read: https://www.merriam-webster.com/grammar/should-you-use-until-or-till-or-til

TLDR:
> So there you have it: you will probably wish to avoid ’till, use ’til advisedly, and use both until and till freely. And if you use till in writing and someone tells you that you have made an error, simply take the extra L off the end of the word and poke them in the eye with it.

Reply:
haha, I admit I wasn't sure how to phrase this and did doubt.

So, I believe this should be?
`Preview unavailable until the data source table has been created`

Reply:
https://github.com/dimagi/commcare-hq/pull/35679/commits/946677f21587cc90d558c0f24a4867c35745befe
Now I can be at peace, this was troubling me.